src/syntax.ml

```diff
@@ -22,12 +22,16 @@ type priority = float
 (** Representation of a binary operator. *)
 type binop = string * assoc * priority * qident
 
+type implicitness =
+  | ImplicitAsDeclared
+  | FullyExplicit
```

Comment:
Why not just Implicit (default) and Explicit (@)?

Reply:
Or remove this definition and use a boolean (`true` if application is explicit).

Reply:
Done, I've used boolean everywhere, even though I find it less clear because you have to remember the meaning of the boolean (and therefore its name). In this case, the boolean is now called "isFullyExplicit".